scripts/test_run_lifecycle_soak.py

```
@@ -505,6 +508,64 @@ def test_every_required_input_proof_is_individually_load_bearing() -> None:
+    def counting_run(command, **keywords):
+        attempts.append(command)
+        return original(command, **keywords)
+
+    MODULE.subprocess.run = counting_run
+    try:
+        failing = [sys.executable, "-c", "raise SystemExit(3)"]
+        try:
+            MODULE.run_checked(failing)
+        except MODULE.SoakError as error:
+            if "3" not in str(error):
+                raise AssertionError("the soak failure did not carry the command's exit status")
```

Ruff (0.16.1)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)